source/agora/common/DNS.d

```diff
@@ -823,17 +848,28 @@ unittest
     ips[0] = InternetAddress.parse(test_ip);
 
     ResourceRecord a_rr = ResourceRecord.make!(TYPE.A)(Domain("localhost"), 0, ips);
+
+    auto uri_target = Address("agora://test.local:1234");
+    ResourceRecord uri_rr = ResourceRecord.make!(TYPE.URI)(
+        Domain("_agora._tcp"), 0, uri_target
```

Comment:
Sorry for not catching that earlier, but why are we using `Domain` here actually ?

Reply:
We don't have to but, RFC states that domains are **extended** in a way that it can accept service labels. We can really extend `Domain` with a, maybe, `Service` type but only difference would be allowing `_` as first character.